u_collate.c

```diff
@@ -28,10 +28,38 @@
 #include <string.h>
 #include <limits.h>
 
+#ifdef __APPLE__
+#include <CoreFoundation/CoreFoundation.h>
+
+/* Helper function for CoreFoundation string comparison */
+static int cf_strcoll(const char *str1, const char *str2, CFStringCompareFlags flags)
+{
+    int result;
+    CFStringRef cfStr1 = CFStringCreateWithCString(NULL, str1, kCFStringEncodingUTF8);
+    CFStringRef cfStr2 = CFStringCreateWithCString(NULL, str2, kCFStringEncodingUTF8);
+    
+    if (cfStr1 && cfStr2) {
+        result = CFStringCompare(cfStr1, cfStr2, flags) - kCFCompareLessThan;
+    } else {
+        /* Fallback to byte comparison if CF conversion fails */
+        result = strcmp(str1, str2);
+    }
+    
+    if (cfStr1) CFRelease(cfStr1);
+    if (cfStr2) CFRelease(cfStr2);
+    
+    return result;
+}
+#endif
+
```

Comment:
Note that `u_strcoll()`/`u_strcasecoll()`/`u_strcasecoll0()` are dead code anyway.  I was thinking about proposing a PR to remove them completely anyway.

Only the `_key` variants of the u_collate.* API is actually used by anything.

So this part of the fix is harmless but also won't actually be called.